### PR TITLE
feat: implement proxy support

### PR DESCRIPTION
**Pull request by rozenmd**

### CODE REVIEW

**Review by petebacondarwin — APPROVED** [review]
Nice simple fix! Approving with a request and a question.

Could you add a bit of explanation to the changeset of how this fixes the problem? Something like, configures the undici library to send all requests via a proxy selected from the first non-empty environment variable from "https_proxy", "HTTPS_PROXY", "http_proxy" and "HTTP_PROXY".

I wonder if we could add a test for this? Since we mock out `fetch` in our unit tests, I don't think it is possible there...

**Comment by rozenmd** [comment]
Thanks @petebacondarwin, I've updated the changeset. 

Also yeah, not entirely sure how we'd test this, still trying to work that out.

**Comment by threepointone** [comment]
It's fine if we don't write this test I think. If we had to, we'd make another fixture app under /examples and set it up there, but it would be slow and not much return for the effort. 

But yay excited to unblock people behind corporate proxies! 

Did you have a look at proxy.ts? I think we may have to do stuff there too. It's fine if it's in a follow up PR.

**Comment by petebacondarwin** [comment]
Good spot @threepointone - I think we do need to fix up to work with a proxy:

```ts
    const remote = connect(`https://${previewToken.host}`);
```

It might be worth creating a manual one-off test setup with a proxy (blocking direct access to CF - if that is not ridiculously hard to do??); and then run through all the Wrangler commands to check we haven't missed anything.

File: packages/wrangler/package.json
Comment (threepointone) [inline packages/wrangler/package.json:99]: Huh that's weird, I thought we were already at 5. Damn, nice catch.

**Comment by rozenmd** [comment]
Closing this PR as I keep finding problems while testing, and I don't want to accidentally merge it.

**Comment by rozenmd** [comment]
Waiting until https://github.com/nodejs/undici/pull/1451 gets released before we can land this PR.

**Pull request merged**
